crates/core/src/operations.rs

```diff
@@ -151,77 +161,60 @@ UPDATE ps_buckets
             // And we need to re-apply all of those.
             // We also replace the checksum with the checksum of the CLEAR op.
             // language=SQLite
-            let clear_statement2 = db.prepare_v2("UPDATE ps_buckets SET last_applied_op = 0, add_checksum = ?1 WHERE name = ?2")?;
+            let clear_statement2 = db.prepare_v2(
+                "UPDATE ps_buckets SET last_applied_op = 0, add_checksum = ?1 WHERE name = ?2",
+            )?;
             clear_statement2.bind_text(2, bucket, sqlite::Destructor::STATIC)?;
             clear_statement2.bind_int(1, checksum)?;
             clear_statement2.exec()?;
+
+            add_checksum = 0;
+            is_empty = true;
         }
     }
 
     if let Some(last_op) = &last_op {
         // language=SQLite
-        let statement = db.prepare_v2("UPDATE ps_buckets SET last_op = ?1 WHERE name = ?2")?;
-        statement.bind_text(2, bucket, sqlite::Destructor::STATIC)?;
-        statement.bind_int64(1, *last_op)?;
-        statement.exec()?;
-    }
-
-
-    // Compact superseded ops immediately
-    if let (Some(first_op), Some(last_op)) = (&first_op, &last_op) {
-        // language=SQLite
-        let statement = db.prepare_v2("UPDATE ps_buckets
```

Comment:
This bit is now largely redundant, since we don't write superseded MOVE/REMOVE ops directly anymore. Technically it is still possible to get a PUT op and supersede it in the same batch, but we just compact those later as part of `clear_remove_ops`.